asf_public_discourse_home_decarbonisation/pipeline/bert_topic_analysis/evaluate_bertopic_results.py

```diff
@@ -0,0 +1,367 @@
+
+    Returns:
+        tuple: data source parameters and model/additional parameters
+    """
+    config_module = importlib.import_module(path_to_config_file)
+
+    data_source_parms = config_module.data_source_parms
+    model_and_additional_params = config_module.model_and_additional_params
+
+    return data_source_parms, model_and_additional_params
+
+
+def create_boxplots_with_results(output_configs: dict):
+    """
+    Creates and saves a figure with boxplots showing the distribution of results for:
+    - Number of topics
+    - Percentage of outliers
+    - Average probability of beloging to a non-outlier topic
+
+    Args:
+        output_configs (dict): dictionary with results information
+    """
+
+    # Extracting information from output_configs dictionary
+    source_name = output_configs["data_source"]
+    category = output_configs["category"]
+    n_docs = output_configs["n_docs"]
+    n_runs = output_configs["n_runs"]
+    keywords = output_configs["keywords"]
+    number_of_outliers_df = output_configs["outliers_df"]
+    number_of_topics_df = output_configs["topics_df"]
+    avg_probablity_df = output_configs["probabilities_df"]
+    outliers_vs_docs_dict = output_configs["outliers_vs_docs_dict"]
+
+    # Plot horizontal boxplots for each dataframe: number_of_topics_df, number_of_outliers_df and avg_probablity_df
+    fig, axes = plt.subplots(3, 1, figsize=(10, number_of_topics_df.shape[1] * 3))
+
+    number_of_topics_df.boxplot(ax=axes[0], grid=False, vert=False)
+    axes[0].set_xlabel("Number of topics")
+
+    percentage_of_outliers_df = number_of_outliers_df / n_docs * 100
+    percentage_of_outliers_df.boxplot(ax=axes[1], grid=False, vert=False)
+    axes[1].set_xlabel("Percentage of outliers")
+    # Making the x-axis always between 0 and 50 (just as visual aid)
+    if percentage_of_outliers_df.max().max() <= 50:
+        axes[1].set_xlim(0, 50)
+
+    avg_probablity_df.boxplot(ax=axes[2], grid=False, vert=False)
+    axes[2].set_xlabel("Average probablity of belonging to a non-outlier topic")
+
+    # axes[3].hist([res["outlier_fraction"] for res in outliers_vs_docs_dict.values()],
+    #     bins=30,
+    #     color=[NESTA_COLOURS[0] for res in outliers_vs_docs_dict.values()],
+    #     edgecolor="white",
+    # )
+    # axes[3].set_xlabel('fraction of runs where a document is an outlier')
+    # axes[3].set_ylabel('# of docs')
+
+    fig.suptitle(
+        "Source: {}, Category: {}\n Keywords filter: {}, # Docs: {}, # Runs: {}".format(
+            source_name, category, keywords, n_docs, n_runs
+        )
+    )
+
+    plt.tight_layout()
+
+    plt.savefig(
+        os.path.join(
+            TOPIC_ANALYSIS_EVALUATION_PATH,
+            f"{source_name}_{category}_{keywords}_{n_runs}runs.png",
+        )
+    )
+
+    plt.close()
+
+
+def run_topic_model_evaluation(n_runs: int, docs: list, model_configs: dict) -> tuple:
+    """
+    Evaluates topic model on a specific number of runs.
```

Comment:
Thinking potentially more of a description of what this function does would be nice. Such as what topic models are being run over?

Reply:
It could also be nice to modularise this function, it's quite big!